sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -445,10 +445,8 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
     stringConcat.toString
   }
 
-  override def sql: String = {
-    val fieldTypes = fields.map(f => s"${quoteIdentifier(f.name)}: ${f.dataType.sql}")
-    s"STRUCT<${fieldTypes.mkString(", ")}>"
-  }
+  override def sql: String =
+    s"STRUCT<${fields.map(_.toDDL.replaceFirst(" ", ": ")).mkString(", ")}>"
```

Comment:
Hm, `replaceFirst(" ", ": ")`. Wouldn't it replace the field that has a space in the middle?

Reply:
Are we allowed to have space in the middle of a field name?

Reply:
Yeah with backquoting, or some places like `toDF`.

```scala
scala> spark.range(1).toDF("a b").schema
res0: org.apache.spark.sql.types.StructType = StructType(StructField(a b,LongType,false))

scala> sql("select 1 as `a b`").schema
res1: org.apache.spark.sql.types.StructType = StructType(StructField(a b,IntegerType,false))
```